### PR TITLE
Update HTML block template

### PR DESCRIPTION
**Pull request by Alfred-Mountfield**
## 🌟 What is the purpose of this PR?

The HTML block template had fallen out of sync with the recent changes. This PR _mostly_ fixes them, although some of the code around the block entity type, and property types, will need updating once we merge `0.3` into main and have stable type URLs on the website.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203623179643287/1204029869185044/f) _(internal)_

## 🚫 Blocked by

N/A

## 🔍 What does this change?

See diff, it's small

## 📜 Does this require a change to the docs?

To my knowledge, no.

## ⚠️ Known issues

- There's a chicken and egg problem with the `esm.sh` part of the template at the moment, it requires a release of the code to be able to point to the package.
- There currently isn't a stable prod URL for types to be hosted at, so I have opted for adjusting the URLs to be "example.com" for now, and modifying the `block-schema` to be an entity type definition. Really the `schema` property of the `block-metadata` should just be the Versioned URL of the schema (which we can have once the changes hit main)
- ~~Trying to run the block template gives a strange error about an object being passed as children to the react component (seems that the props aren't being correctly inserted into `MockBlockDock`)~~ this was caused by our inline comments..

## 🐾 Next steps

Fixup the schema and types definitions, and esm.sh link, when changes hit `main`

## 🛡 What tests cover this?

Unsure if any

## ❓ How to test this?

- Currently it's broken, see known issues above

### CODE REVIEW

File: libs/block-template-html/block-metadata.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
   },
   "protocol": "0.2",
   "displayName": "Template",
-  "examples": [{ "name": "World" }],
+  "examples": [{ "http://example.com/types/property-type/name/": "World" }],
```
Comment (CiaranMn) [inline libs/block-template-html/block-metadata.json:14]: Could use the same as in our other templates, which is a real URL

File: libs/block-template-html/block-schema.json [hunk 2]
```diff
@@ -1,11 +1,14 @@
 {
+  "$id": "http://example.com/types/entity-type/example/v/1",
```
Comment (CiaranMn) [inline libs/block-template-html/block-schema.json:2]: Ditto other comment – could use `Thing` from the other templates

File: libs/block-template-html/block-schema.json [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
   "properties": {
-    "name": {
-      "type": "string"
+    "http://example.com/types/property-type/name/": {
+      "$ref": "http://example.com/types/property-type/name/v/1"
     }
   },
-  "required": ["name"],
+  "required": ["http://example.com/types/property-type/name/"],
+  "additionalProperties": false,
```
Comment (CiaranMn) [inline libs/block-template-html/block-schema.json:12]: 🧐

File: libs/block-template-html/block-schema.json (hunk 2 above)
Comment (CiaranMn) [reply libs/block-template-html/block-schema.json:2]: In fact we should probably just delete this file, it is a relic from when blocks bundled their schemas

File: libs/block-template-html/block-metadata.json (hunk 1 above)
Comment (Alfred-Mountfield) [reply libs/block-template-html/block-metadata.json:14]: Happy to do that now

File: libs/block-template-html/block-metadata.json [hunk 4]
```diff
@@ -11,7 +11,7 @@
   },
   "protocol": "0.2",
```
Comment (CiaranMn) [inline libs/block-template-html/block-metadata.json:12]:
```suggestion
  "protocol": "0.3",
```

Comment (Alfred-Mountfield) [reply libs/block-template-html/block-metadata.json:12]: Should we even have this file or should it be in `package.json` now?

Comment (CiaranMn) [reply libs/block-template-html/block-metadata.json:12]: This has a different bundling script which doesn't do the 'grab things from package.json'. so we potentially could but would need to update the bundling to generate this file

Comment (Alfred-Mountfield) [reply libs/block-template-html/block-metadata.json:12]: Gotcha, keep the file for now then

Comment (Alfred-Mountfield) [reply libs/block-template-html/block-metadata.json]: [Remove redundant files, use real types](https://github.com/blockprotocol/blockprotocol/pull/1021/commits/20ab0d7034706eb82f4e916d10e57f3563a5387e)

File: libs/block-template-html/block-metadata.json (hunk 1 above)
Comment (Alfred-Mountfield) [reply libs/block-template-html/block-metadata.json]: [Remove redundant files, use real types](https://github.com/blockprotocol/blockprotocol/pull/1021/commits/20ab0d7034706eb82f4e916d10e57f3563a5387e)

**Review by CiaranMn — APPROVED** [review]

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**